ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -66,6 +66,7 @@ type ContainerDef struct {
 	MemoryReservation libYaml.MemStringorInt `yaml:"mem_reservation"`
 	HealthCheck       *HealthCheck           `yaml:"healthcheck"`
 	Secrets           []Secret               `yaml:"secrets"`
+	GPU               string                 `yaml:"gpu"`
```

Comment:
In the container definition, GPU resources are specified within a list of ['resourceRequirement' objects](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ResourceRequirement.html), vs. as a flat value. If new `resourceRequirement` types are added in the future, how do you propose we add them? Flattened as well? What if multiple requirements of the same type are allowed? Wondering what your ideas are here re: keeping this extensible.

One alternative would be to take in a list of these, like we do for secrets:
```
secrets:
        - value_from: string
          name: string
...
resource_requirements:
       - type: GPU
         value: 3
```

Reply:
+1 was about to make the same comment myself

Reply:
Oh wait, ask @efekarakus to explain this offline, he has an explanation for why this is the preferred approach.

Reply:
Discussed offline 😝

Reply:
Looks like this was resolved, but Efe was following a design choice I made in a previous design doc.